Tests/Tracking/CustomerIOImplementationTest.swift

```diff
@@ -426,4 +428,56 @@ class CustomerIOImplementationTest: UnitTest {
         XCTAssertEqual(actualQueueTaskData?.event, givenEvent)
         XCTAssertEqual(actualQueueTaskData?.deviceToken, givenDeviceToken)
     }
+
+    // MARK: handleQueueBacklog/getAndProcessTask
+
+    func test_givenEmptyBacklog_expectNoTasksProcessed() {
+        backgroundQueueMock.getAllStoredTasksReturnValue = []
+        XCTAssertNotNil(implementation.handleQueueBacklog())
+        XCTAssertEqual(backgroundQueueMock.getAllStoredTasksCallsCount, 1)
+    }
+
+    func test_givenBacklog_expectTaskProcessed() {
+        var inventory: [QueueTaskMetadata] = []
+        let givenType = QueueTaskType.identifyProfile
+        let givenTask = IdentifyProfileQueueTaskData(identifier: String.random, attributesJsonString: "null")
```

Comment:
we are only appending the same kind of task, it would have been great if we could have an additional test case where we verify each type of task and make sure its values are being received accordingly.

Reply:
I do not think that will change the functionality in any way., Since it is a test case and not actual implementation so I do not believe it will make any change!